util/lambda_trigger/releases.go

```diff
@@ -1,21 +1,54 @@
 package main
 
 import (
+	"encoding/json"
 	"fmt"
-	"net/url"
-
-	hb "github.com/gulducat/hashi-bin/types"
+	"net/http"
+	"strings"
 )
 
-// ReleasesURL The url to read the releases index from
-const ReleasesURL = "https://releases.hashicorp.com"
+type ReleaseResponse Release
+
+type Release struct {
+	Version string
+}
+
+func releasesAPIRequest(apiBaseURL, productName string, license_class string) (*string, error) {
+	release := ReleaseResponse{}
+	url := fmt.Sprintf("%s/%s/latest", apiBaseURL, productName)
+	// For ENT products, set the license_class parameter to filter by ENT-only
+	if license_class == "enterprise" {
+		productName = strings.TrimSuffix(productName, "-enterprise")
+		url = fmt.Sprintf("%s/%s/latest?license_class=enterprise", apiBaseURL, productName)
+	}
 
-func getLatestVersion(productName string) (*hb.Version, error) {
-	safeProduct := url.PathEscape(productName)
-	url := fmt.Sprintf("%s/%s/%s", ReleasesURL, safeProduct, "index.json")
-	product, err := hb.NewProduct(url)
+	fmt.Printf("Reading %s", url)
+	resp, err := http.Get(url)
 	if err != nil {
-		return &hb.Version{}, err
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if err := json.NewDecoder(resp.Body).Decode(&release); err != nil {
+		return nil, err
 	}
-	return product.LatestVersion(), nil
+
+	return &release.Version, nil
```

Comment:
At this point, you know the request got a response, but the response data could be an error object like
```json
{"message":"release not found","code":30006}
```
The `http.Get` function doesn't return an error for non-2xx responses.  Callers assume that checking `err != nil` is sufficient (lambda_trigger.go above), so this function should check if the response indicates an error or lacks a valid version string, and if so create an appropriate response error for the caller to check.

Reply:
Ah, I just forgot to put in that last error check. Thanks for the pointer!

Reply:
Ok, I added this into the latest commit and tested with `consul-dataplane` and `vault`. [Consul dataplane is one that I'll need to manually bump once they release their next stable version, since the API doesn't set `latest` for any pre-releases] 

<img width="649" alt="Screen Shot 2022-10-03 at 2 02 02 PM" src="https://user-images.githubusercontent.com/12939567/193682141-03c2e338-b86d-4cb5-a2ee-91a5a9693004.png">

<img width="700" alt="Screen Shot 2022-10-03 at 2 02 59 PM" src="https://user-images.githubusercontent.com/12939567/193682143-4e15c1af-f0de-4834-9c3f-2df539bc5287.png">